jina/math/dimensionality_reduction.py

```diff
@@ -0,0 +1,58 @@
+import numpy as np
+
+
+class PCA:
+    """A class for PCA dimensionality reduction in Jina."""
+
+    def __init__(self, n_components: int, whitening: bool = False):
+        self.n_components = n_components
+        self.whitening = whitening
```

Comment:
if my memery is correct, whitening should be always true?

Reply:
sklearn does not do it by default, I just made an implementation that is equivalent to the default sklearn version